### ISSUE TITLE
请问TCLoss代码什么时候上传

### ISSUE THREAD

**Issue by hcqylymzc** [YaoleiQi/DSCNet#10]
没有在code里找到

**Comment by T0kisaki-Kurumi** [comment]
After analyzing the code and paper in the past few days, I have found that this loss function is likely not implemented by the author himself, and in more serious cases, the paper may be suspected of fraud (a speculation). The publication of such papers on ICCV is simply a pollution to the academic environment ...

**Comment by YaoleiQi** [comment]
Thank you very much for your review of our code and paper. In response to your serious question, we would also like to engage in a serious discussion about whether this innovation has any issues:

1.First and foremost, let's address the most serious concern, the 'plagiarism issue': The use of PH in designing the loss is not our original contribution, and this is explicitly stated in the paper. We have also conducted comparisons with representative works in the field, and the experimental results presented in the paper are based on genuine implementations."

2.Regarding the innovation points: Firstly, the core innovation of our work lies in the design of novel dynamic snake kernels. This part of the code was personally developed by me, in collaboration with my team members. I would like to know where any potential plagiarism is suspected in this section.

3.Secondly, in terms of the innovative loss functions, as mentioned in our paper and text, we have discussed the issues with existing methods in real-world scenarios, including problems with discrete points. We believe that papers are meant for sharing and discussing, and we have presented our proposed solutions. We have attempted to devise strategies to address these challenges and have applied them in 3D medical data. Using PH-based loss functions in 3D tasks with existing methods can be highly costly, and we have made optimizations to our code. Based on our own insights and efforts, I cannot accept any questioning of our work.

4.Lastly, and most importantly, with regards to open-sourcing the code, I am currently a dedicated research-oriented student with a strong interest in scientific research. While our method may not be the absolute best, it is undoubtedly applicable in clinical and real-world tasks. Currently, we have not open-sourced the code for the loss function due to two reasons:

1)My advisor has strictly advised me to slow down the process of open-sourcing this particular code. This is not a decision I can make unilaterally. As a student, I do not have complete decision-making authority, and I sincerely apologize for this.

2)I am presently involved in organizing open datasets and providing some pre-trained parameters to assist everyone in using our model effectively. This leaves me with limited time to focus on code encapsulation and open-sourcing. I also hope to meet everyone through a challenge format, working together to advance academic development. I believe that, by that time, I can convince my professor.

If you have any more questions, I am more than willing to have a serious discussion with you. We do not claim to be the best in the field. As long as one aspect of our work can inspire many people, and I open-source it for free for everyone to use, we have received acknowledgments from many colleagues who have achieved excellent results in their respective fields using our method. Can this be considered contamination?

**Comment by YaoleiQi** [comment]
If it's convenient, we can connect on WeChat: 13082556710, or schedule some meetings, for example, using Tencent Meeting, Zoom, or other platforms to discuss any concerns you might have. We can compare, reproduce under your supervision, or explore other ways. It's challenging when work you've put sincere effort into gets questioned; it's natural to feel uncomfortable. We hoped our work could be valuable and applicable elsewhere. Additionally, we hope you can take responsibility for your statements. Thank you once again for your feedback.

**Comment by T0kisaki-Kurumi** [comment]
> Thank you very much for your review of our code and paper. In response to your serious question, we would also like to engage in a serious discussion about whether this innovation has any issues:
> 
> 1.First and foremost, let's address the most serious concern, the 'plagiarism issue': The use of PH in designing the loss is not our original contribution, and this is explicitly stated in the paper. We have also conducted comparisons with representative works in the field, and the experimental results presented in the paper are based on genuine implementations."
> 
> 2.Regarding the innovation points: Firstly, the core innovation of our work lies in the design of novel dynamic snake kernels. This part of the code was personally developed by me, in collaboration with my team members. I would like to know where any potential plagiarism is suspected in this section.
> 
> 3.Secondly, in terms of the innovative loss functions, as mentioned in our paper and text, we have discussed the issues with existing methods in real-world scenarios, including problems with discrete points. We believe that papers are meant for sharing and discussing, and we have presented our proposed solutions. We have attempted to devise strategies to address these challenges and have applied them in 3D medical data. Using PH-based loss functions in 3D tasks with existing methods can be highly costly, and we have made optimizations to our code. Based on our own insights and efforts, I cannot accept any questioning of our work.
> 
> 4.Lastly, and most importantly, with regards to open-sourcing the code, I am currently a dedicated research-oriented student with a strong interest in scientific research. While our method may not be the absolute best, it is undoubtedly applicable in clinical and real-world tasks. Currently, we have not open-sourced the code for the loss function due to two reasons:
> 
> 1)My advisor has strictly advised me to slow down the process of open-sourcing this particular code. This is not a decision I can make unilaterally. As a student, I do not have complete decision-making authority, and I sincerely apologize for this.
> 
> 2)I am presently involved in organizing open datasets and providing some pre-trained parameters to assist everyone in using our model effectively. This leaves me with limited time to focus on code encapsulation and open-sourcing. I also hope to meet everyone through a challenge format, working together to advance academic development. I believe that, by that time, I can convince my professor.
> 
> If you have any more questions, I am more than willing to have a serious discussion with you. We do not claim to be the best in the field. As long as one aspect of our work can inspire many people, and I open-source it for free for everyone to use, we have received acknowledgments from many colleagues who have achieved excellent results in their respective fields using our method. Can this be considered contamination?

I'm sorry for posting an abrupt comment on Github. Currently, I have deleted the comment to avoid any negative impact. I fully understand that it is not your obligation to upload the loss function code, but I have learned from your previous replies to others that you currently do not have a good way to reduce complexity, so I will not further explore this issue.

Once again, I apologize for my irresponsible comments. It is difficult to assess the effectiveness of a job in today's academic paper flooding situation, but I also deeply apologize for my unfounded speculation.

**Comment by YaoleiQi** [comment]
No problem, and thank you very much! there are indeed some practical reasons that make it a bit tricky, and I've been frustrated as well. 

But everyone has the right and freedom to express their opinions and preferences about articles, and I respect that! 

I also appreciate finding colleagues who are dedicated to serious research. If there's anything you'd like to discuss, we can connect as friends, and I hope we can collaborate in the future!
